### PR TITLE
Add support for GWT

### PR DESCRIPTION
**Pull request by laurit**
Add span around java method called from javascript.

### CODE REVIEW

File: instrumentation/gwt-2.0/javaagent/gwt-2.0-javaagent.gradle [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  testImplementation "org.testcontainers:selenium:${versions.testcontainers}"
+  testImplementation 'org.seleniumhq.selenium:selenium-java:3.141.59'
+
+  testImplementation(project(':testing-common')) {
+    exclude group: 'org.eclipse.jetty', module: 'jetty-server'
+  }
+  testImplementation group: 'org.eclipse.jetty', name: 'jetty-webapp', version: '9.4.35.v20201120'
+}
+
+task copyTestWebapp(type: Copy) {
```
Comment (iNikem) [inline instrumentation/gwt-2.0/javaagent/gwt-2.0-javaagent.gradle:48]: Wouldn't it be easier to have a separate module for test app?

**Review by iNikem — APPROVED** [review]
Anything we can do for SERVER span name?

File: instrumentation/gwt-2.0/javaagent/gwt-2.0-javaagent.gradle (hunk 1 above)
Comment (laurit) [reply instrumentation/gwt-2.0/javaagent/gwt-2.0-javaagent.gradle:48]: Assuming that we want to compile the test app with the same version of gwt that we use to run the test even if it is doable having the test app in another module would probably make this even more confusing.

Comment (laurit) [reply instrumentation/gwt-2.0/javaagent/gwt-2.0-javaagent.gradle:48]: As for the server span name suggestions are welcome. Currently it is the servlet mapping where the service is mapped as far as I can tell each service should be mapped to a different path so it should be a rather good name. Alternative would be to set server span to invoked method name, but then server span and nested span would have the same name.

File: instrumentation/gwt-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/gwt/GwtTracer.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+package io.opentelemetry.javaagent.instrumentation.gwt;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import java.lang.reflect.Method;
+
+public class GwtTracer extends BaseTracer {
+  private static final ContextKey<Object> PRC_CONTEXT_KEY =
```
Comment (mateuszrzeszutek) [inline instrumentation/gwt-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/gwt/GwtTracer.java:15]: Typo:
```suggestion
  private static final ContextKey<Object> RPC_CONTEXT_KEY =
```

File: instrumentation/gwt-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/gwt/GwtInstrumentationModule.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+      Map<ElementMatcher<MethodDescription>, String> transformers = new HashMap<>();
+
+      transformers.put(
+          named("invokeAndEncodeResponse")
+              .and(takesArguments(5))
+              .and(takesArgument(0, named(Object.class.getName())))
+              .and(takesArgument(1, named(Method.class.getName())))
+              .and(takesArgument(3, named("com.google.gwt.user.server.rpc.SerializationPolicy")))
+              .and(takesArgument(4, named(int.class.getName()))),
```
Comment (mateuszrzeszutek) [inline instrumentation/gwt-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/gwt/GwtInstrumentationModule.java:64]: There's no need for `named()` matcher when referencing standard Java types:
```suggestion
              .and(takesArgument(0, Object.class))
              .and(takesArgument(1, Method.class))
              .and(takesArgument(3, named("com.google.gwt.user.server.rpc.SerializationPolicy")))
              .and(takesArgument(4, int.class)),
```

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/gwt-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/gwt/GwtInstrumentationModule.java (hunk 3 above)
Comment (laurit) [reply instrumentation/gwt-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/gwt/GwtInstrumentationModule.java]: Thanks, didn't know that. Should be fixed now.

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**